### DISCUSSION TITLE
How do you turn off the OpenTelementry INFO messages in Rails startup?

### DISCUSSION THREAD

**Discussion by rnhurt** [open-telemetry/opentelemetry-ruby-contrib#1165] [Q&A] [answered]
When we turned on OpenTelemetry in our Rails 6/7 apps we started getting a bunch of INFO messages dumped to stdout.  Nothing else in our Rails apps generates these types of messages and I can't seem to find a way to turn them off.  They don't seem to obey the Rails.logger config settings.

Is anyone else having this problem or is it just me?  Am I missing some config somewhere that disables these INFO messages?  Or is this a bug in OpenTelementry?

```
...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::ActiveSupport was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::Rack was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::ActionPack was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::ActiveJob was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::ActiveRecord was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::ActionView was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::ActionMailer was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::HTTP was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::ActiveModelSerializers was successfully ...
I, [2024-09-13] INFO -- : Instrumentation: OpenTelemetry::Instrumentation::ConcurrentRuby was successfully ...
...
```

**Comment by arielvalentin** [comment, marked as answer]
The most consistent way to control log verbosity for the OTel SDK is to use the `OTEL_LOG_LEVEL` environment variable, which is set to `INFO` by default[^1].

You may also provide your own logger to the SDK Configurator during initialization[^2]. 

E.g.

```ruby
OpenTelemetry::SDK.configure do |c|
  c.logger = rails_configuration.logger  
  # custom configuration
end
```

In this case the OTel logger will _wrap_ your custom logger and send it messages at `info` level, unless you set the envar.

From there your logger will receive `info` or higher.

[^1]: https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration 
[^2]: https://www.rubydoc.info/gems/opentelemetry-sdk/OpenTelemetry/SDK/Configurator#logger=-instance_method

**Reply by rnhurt** [reply]
Yes, that seems to work.  

Thanx!  🎉

**Reply by robbkidd** [reply]
In the near future, the OTel SDK will provide the capability to export logs from the Rails logger as OpenTelemetry data. At that point, you might consider _not_ using the same logger instances for the OTel SDK as for the rest of your application. The OTel SDK's log output is where errors around sending data will appear … and if the OTel SDK is having trouble sending data, the logs telling you about that trouble will be among the data _not_ being sent.
